src/test/java/org/gridsuite/loadflow/server/LoadFlowServiceTest.java

```diff
@@ -0,0 +1,99 @@
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+class LoadFlowServiceTest {
+
+    @MockBean
+    private LimitViolationRepository limitViolationRepository;
+
+    @Autowired
+    private LoadFlowService loadFlowService;
+
+    private static final UUID RESULT_UUID = UUID.randomUUID();
+
+    private static final class LimitViolationsMock {
+        static List<LimitViolationEntity> limitViolationEntities = Arrays.asList(
+                LimitViolationEntity.builder()
+                        .id(UUID.randomUUID())
+                        .loadFlowResult(LoadFlowResultEntity.builder().resultUuid(RESULT_UUID).build())
+                        .subjectId("Subject1")
+                        .limit(100.0)
+                        .limitName("Voltage")
+                        .actualOverload(110)
+                        .upComingOverload(120)
+                        .overload(10.0)
+                        .value(110.0)
+                        .side("A")
+                        .limitType(LimitViolationType.CURRENT)
+                        .build(),
+                LimitViolationEntity.builder()
+                        .id(UUID.randomUUID())
+                        .loadFlowResult(LoadFlowResultEntity.builder().resultUuid(RESULT_UUID).build())
+                        .subjectId("Subject2")
+                        .limit(200.0)
+                        .limitName("Current")
+                        .actualOverload(210)
+                        .upComingOverload(220)
+                        .overload(20.0)
+                        .value(210.0)
+                        .side("B")
+                        .limitType(LimitViolationType.CURRENT)
+                        .build()
+        );
+    }
+
+    @Test
+    void assertResultExistsWhenResultExistsShouldNotThrowException() {
+        UUID existingUuid = UUID.randomUUID();
+        when(limitViolationRepository.existsLimitViolationEntitiesByLoadFlowResultResultUuid(existingUuid)).thenReturn(true);
+        assertDoesNotThrow(() -> loadFlowService.assertResultExists(existingUuid));
+    }
+
+    @Test
+    void assertResultExistsWhenResultDoesNotExistShouldThrowLoadflowException() {
+        UUID nonExistingUuid = UUID.randomUUID();
+        when(limitViolationRepository.existsLimitViolationEntitiesByLoadFlowResultResultUuid(nonExistingUuid)).thenReturn(false);
+        LoadflowException thrown = assertThrows(LoadflowException.class, () -> loadFlowService.assertResultExists(nonExistingUuid));
+        assertEquals(LoadflowException.Type.RESULT_NOT_FOUND, thrown.getType());
+    }
+
+    @Test
+    void getLimitViolationsInfosWhenCalledReturnsCorrectData() {
+
+        String stringFilters = "[{\"column\":\"subjectId\",\"dataType\":\"text\",\"type\":\"startsWith\",\"value\":\"Sub\"}]";
```

Comment:
may be i didn't understand everything, but this test is not complete in my opinion, all objects in the data set start with "Sub"
may be add a some data that we expect to be filtred ?